autonomy/cli/scaffold_fsm.py

```diff
@@ -643,6 +647,12 @@ def skill_test_dir(self) -> Path:
         """Get the directory to the skill tests."""
         return self.skill_dir / "tests"
 
+    @classmethod
+    def _parse_spec(cls, spec_path: Path) -> DFA:
+        """Parse the FSM specification."""
+        with spec_path.open(encoding="utf-8") as fp:
+            return DFA.load(fp, input_format="yaml")
+
```

Comment:
this is bloat. It should not be a classmethod either. It can be replaced entirely with 
`self.dfa = DFA.load(spec_path.read_text(), input_format="yaml")` in the `__init__`. Please don't use class methods when not needed, and avoid the use of classes all together whenever possible - this method could be a function, were it not the case that it should not exist to begin with.